tests/network_test.go

```diff
@@ -0,0 +1,105 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta2"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	v12 "k8s.io/api/core/v1"
+	"testing"
+)
+
+/*
+	Test that `hostNetwork` in SparkApplication propagates to driver and executor pods
+*/
+func TestHostNetworkPropagation(t *testing.T) {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var testCases = []struct {
+		driverHN   bool
+		executorHN bool
+	}{
+		{false, false},
+		{true, false},
+		{false, true},
+		{true, true},
+	}
+
+	for i, tc := range testCases {
+		log.Infof("Running test case:\n- driver host network:\t\t%v\n- executor host network:\t%v", tc.driverHN, tc.executorHN)
+		jobName := fmt.Sprintf("host-network-test-job-%d", i)
+		job := utils.SparkJob{
+			Name:     jobName,
+			Template: "spark-mock-task-runner-job-host-network.yaml",
+			Params: map[string]interface{}{
+				"args":                []string{"1", "600"},
+				"driverHostNetwork":   tc.driverHN,
+				"executorHostNetwork": tc.executorHN,
+			},
+		}
+
+		expectedExecutorCount := 1
+
+		// Submit the job and wait for it to start
+		err = spark.SubmitJob(&job)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = spark.WaitForJobState(job, v1beta2.RunningState)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Wait for correct number of executors to show up
+		err = utils.Retry(func() error {
+			executors, err := spark.GetExecutorState(job)
+			if err != nil {
+				return err
+			} else if len(executors) != expectedExecutorCount {
+				return errors.New(fmt.Sprintf("The number of executors is %d, but %d is expected", len(executors), expectedExecutorCount))
+			}
+			return nil
+		})
+		if err != nil {
+			t.Error(err)
+		}
```

Comment:
this part looks generalizable and applicable to all the tests where jobs are submitted. Can we extract this logic in some generic `SubmitAndWait` or alike?

Reply:
Make sense. Created a method for it.